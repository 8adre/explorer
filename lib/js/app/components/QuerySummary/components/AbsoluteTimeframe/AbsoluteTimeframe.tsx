import React, { FC } from 'react';
import moment from 'moment-timezone';

import { getTimezoneValue } from '../../../../queryCreator';
import { Container, Separator } from './AbsoluteTimeframe.styles';

import { Timezones } from '../../../../queryCreator';

import text from './text.json';

type Props = {
  timeframe: {
    start: string;
    end: string;
  };
  timezone: Timezones | number;
};

const AbsoluteTimeframe: FC<Props> = ({ timeframe, timezone }) => {
  const { start, end } = timeframe;
  const namedTimezone = getTimezoneValue(timezone);

  return (
    <Container>
      <span>{moment(start).tz(namedTimezone).format('YYYY-MM-DD HH:mm')}</span>
      <Separator>{text.separator}</Separator>
      <span>{moment(end).tz(namedTimezone).format('YYYY-MM-DD HH:mm')}</span>
    </Container>
  );
};

export default AbsoluteTimeframe;
