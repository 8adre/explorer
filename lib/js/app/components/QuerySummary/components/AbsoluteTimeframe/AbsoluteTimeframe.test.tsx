import React from 'react';
import { render as rtlRender } from '@testing-library/react';

import AbsoluteTimeframe from './AbsoluteTimeframe';

const render = (overProps: any = {}) => {
  const props = {
    timezone: 'UTC',
    timeframe: {
      start: '2020-09-29T00:00:00Z',
      end: '2020-09-30T00:00:00Z',
    },
    ...overProps,
  };

  const wrapper = rtlRender(<AbsoluteTimeframe {...props} />);

  return {
    wrapper,
    props,
  };
};

test('renders <AbsoluteTimeframe /> component', () => {
  const {
    wrapper: { container },
  } = render();

  expect(container).toMatchSnapshot();
});

test('renders <AbsoluteTimeframe /> component when timezone as number provided', () => {
  const {
    wrapper: { container },
  } = render({ timezone: 0 });

  expect(container).toMatchSnapshot();
});
