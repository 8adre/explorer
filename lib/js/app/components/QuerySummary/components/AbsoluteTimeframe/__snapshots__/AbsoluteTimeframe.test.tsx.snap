// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <AbsoluteTimeframe /> component 1`] = `
<div>
  <div
    class="sc-pYBma bXhNpl"
  >
    2020-09-29 00:00
    <span
      class="sc-oUbqx eulMNf"
    >
      to
    </span>
    2020-09-30 00:00
  </div>
</div>
`;

exports[`renders <AbsoluteTimeframe /> component when timezone as number provided 1`] = `
<div>
  <div
    class="sc-pYBma bXhNpl"
  >
    2020-09-29 00:00
    <span
      class="sc-oUbqx eulMNf"
    >
      to
    </span>
    2020-09-30 00:00
  </div>
</div>
`;
